javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -526,30 +547,22 @@ private Vector(int indexShift, HashArrayMappedTrie<Integer, T> trie) {
     public Vector<T> drop(long n) {
         if (n <= 0) {
             return this;
-        }
-        if (n >= length()) {
+        } else if (n >= length()) {
```

Comment:
Just because the length is constant at the moment, doesn't mean it won't change (i.e. it's an implementation detail that may change, and we don't want to update it later everywhere, just in the `length` method)